Sources/MongoSwift/MongoCollection+Indexes.swift

```diff
@@ -258,35 +258,33 @@ extension MongoCollection {
             throw getErrorFromReply(bsonError: error, from: reply)
         }
 
-        return forModels.map { $0.options?.name ?? $0.defaultName }
+        return models.map { $0.options?.name ?? $0.defaultName }
     }
 
     /**
      * Drops a single index from the collection by the index name.
      *
      * - Parameters:
      *   - name: The name of the index to drop
-     *   - writeConcern: An optional WriteConcern to use for the command
+     *   - options: Optional `DropIndexOptions` to use for the command
      *
      * - Throws:
+     *   - `ServerError.writeError` if an error occurs while performing the command.
      *   - `ServerError.commandError` if an error occurs that prevents the command from executing.
      *   - `UserError.invalidArgumentError` if the options passed in form an invalid combination.
+     *   - `EncodingError` if an error occurs while encoding the options.
      */
-    public func dropIndex(_ name: String, options: DropIndexOptions? = nil) throws {
-        let opts = try self.encoder.encode(options)
-        var error = bson_error_t()
-        guard mongoc_collection_drop_index_with_opts(self._collection, name, opts?.data, &error) else {
-            throw parseMongocError(error)
-        }
+    @discardableResult
```

Comment:
we had this returning nothing, but according to the spec should return same result as all the other methods.